dev/RadioMenuFlyoutItem/RadioMenuFlyoutItem.cpp

```diff
@@ -50,31 +67,51 @@ void RadioMenuFlyoutItem::OnInternalIsCheckedChanged(const winrt::DependencyObje
     else if (!IsChecked())
     {
         IsChecked(true);
-        UpdateSiblings();
+        UncheckPreviousIfNecessary();
     }
 }
 
-void RadioMenuFlyoutItem::UpdateSiblings()
+void RadioMenuFlyoutItem::UncheckPreviousIfNecessary()
 {
     if (IsChecked())
     {
-        // Since this item is checked, uncheck all siblings
-        if (auto parent = winrt::VisualTreeHelper::GetParent(*this))
+        const auto groupName = GroupName();
+
+        if (s_selectionMap.HasKey(groupName))
         {
-            const int childrenCount = winrt::VisualTreeHelper::GetChildrenCount(parent);
-            for (int i = 0; i < childrenCount; i++)
+            if (const auto previousCheckedItem = s_selectionMap.Lookup(groupName))
             {
-                auto child = winrt::VisualTreeHelper::GetChild(parent, i);
-                if (auto radioItem = child.try_as<winrt::RadioMenuFlyoutItem>())
+                // Uncheck the previously checked item.
+                previousCheckedItem.IsChecked(false);
+            }
+        }
+        s_selectionMap.Insert(groupName, *this);
+    }
+}
+
+void RadioMenuFlyoutItem::OnAreCheckStatesEnabledPropertyChanged(const winrt::DependencyObject& sender, const winrt::DependencyPropertyChangedEventArgs& args)
```

Comment:
>OnAreCheckStatesEnabledPropertyChanged [](http://example.com/codeflow?start=26&length=38)

the property could also be set after the loaded event is called, in which case we will skip this. Not sure if there is a valid scenario there though.

Reply:
I don't really expect this -- it would be pretty weird to change the Style of the submenu while it's open.